hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1196,28 +1195,60 @@ public void reset() {
     @Override
     public void destroy() {
         clearPartition(false, true);
-        storage.destroy(false);
-        mutationObserver.onDestroy(false);
     }
 
     @Override
-    public void destroyInternals() {
-        clearMapStore();
-        clearStorage(false);
-        storage.destroy(false);
-        mutationObserver.onDestroy(true);
+    public void clearPartition(boolean onShutdown, boolean onStorageDestroy) {
+        clearOtherResourcesThanStorage(onStorageDestroy);
```

Comment:
There's a bit of a difference between the old behaviour and the new and I'm not sure if it's ok. Analysing by different parameters:

- onShutdown false, onRecordStoreDestroy false
The code stays the same.



- onShutdown false, onRecordStoreDestroy true
Previously:
```
        if (hasPooledMemoryAllocator()) {
            storage.clear(true);
            mutationObserver.onClear();
        }
        storage.destroy(true);
        mutationObserver.onDestroy(true);
```

Now:
```
        storage.clear(false);
        mutationObserver.onClear();
        storage.destroy(false);
        mutationObserver.onDestroy(false);
```
Some changes: we clear the store even if the allocator is not POOLED and we clear the hot restart store, previously we haven't.


- onShutdown true, onRecordStoreDestroy false
Previously:
```
        if (hasPooledMemoryAllocator()) {
            storage.clear(true);
            mutationObserver.onClear();
        }
        storage.destroy(true);
        mutationObserver.onDestroy(true);
```

Now:
```
        if (hasPooledMemoryAllocator()) {
            storage.destroy(true);
            mutationObserver.onDestroy(true);
        } else {
            storage.clear(true);
            mutationObserver.onClear();
            storage.destroy(true);
            mutationObserver.onDestroy(true);
        }
```
Changes: Previously we cleared and destroyed the store when the allocator is POOLED, now we just destroy it. If the allocator is not POOLED, we additionally clear.


- onShutdown true, onRecordStoreDestroy true
Previously:
```
        if (hasPooledMemoryAllocator()) {
            storage.clear(true);
            mutationObserver.onClear();
        }
        storage.destroy(true);
        mutationObserver.onDestroy(true);
```

Now:
```
        if (hasPooledMemoryAllocator()) {
            storage.destroy(true);
            mutationObserver.onDestroy(true);
        } else {
            storage.clear(true);
            mutationObserver.onClear();
            storage.destroy(true);
            mutationObserver.onDestroy(true);
        }
```
Changes are the same as for onShutdown true, onRecordStoreDestroy false.

Those are quite a few changes. Can you confirm this is ok?

Reply:
👍

Reply:
Double-checked this comment, seems ok. Thanks for the explanation.